.travis.yml

```diff
@@ -1,45 +1,71 @@
-language: python
+language: c
 
 env:
-  - PYTHON_VERSION=3.6
+  - PYTHON_VERSION=3.7
 
-addons:
-  apt:
-    sources:
-    - ubuntu-toolchain-r-test
-    packages:
-    - libopenmpi-dev
+matrix:
+  include:
+      # Linux
+      - os: linux
+        name: "Linux full"
+        addons:
+          apt:
+            sources:
+              - ubuntu-toolchain-r-test
+            packages:
+              - libopenmpi-dev
+              - openmpi-bin
+      # OSX
+      - os: osx
+        name: "MacOS full"
 
 
 before_install:
-  - wget -q http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh
+  - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
+        wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+    else
+        wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O miniconda.sh;
+    fi;
   - chmod +x miniconda.sh
-  - ./miniconda.sh -b -p /home/travis/miniconda
-  - export PATH=/home/travis/miniconda/bin:$PATH
+  - ./miniconda.sh -b -p ${HOME}/miniconda
+  - export PATH=${HOME}/miniconda/bin:$PATH
   - conda update --yes --quiet conda
 
 install:
     - install_prefix=~
     - conda create -n testenv --yes pip python=${PYTHON_VERSION}
     - source activate testenv
     - conda install --yes scipy numpy matplotlib
+    - |
+      if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+        conda install --yes mpi4py openmpi
+      else
+        pip install mpi4py
+      fi
     - pip install flake8 pytest pytest-cov
-    - pip install mpi4py mne
+    - pip install mne psutil joblib
     - pip install coverage coveralls
+    - travis_wait 60
     - |
-      git clone https://github.com/neuronsimulator/nrn
-      cd nrn
-      ./build.sh
-      ./configure --with-nrnpython=python3 --without-iv --prefix=${install_prefix}
-      make -j4
-      make install -j4
-      cd src/nrnpython/
-      python3 setup.py install
-    - |
-      cd $TRAVIS_BUILD_DIR
-      export PATH=$PATH:${install_prefix}/x86_64/bin
-      make
-      python setup.py develop
+      if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+        wget -q  https://neuron.yale.edu/ftp/neuron/versions/v7.7/nrn-7.7.x86_64-osx.pkg -O nrn.pkg
+        sudo installer -pkg nrn.pkg -allowUntrusted -target /
+      else
+        git clone https://github.com/neuronsimulator/nrn
+        cd nrn
+        ./build.sh
+        ./configure --with-nrnpython=python3 --without-iv --prefix=${install_prefix} \
+          --with-paranrn --disable-rx3d --without-nrnoc-x11 --with-mpi
+        make -j4
+        make install -j4
+        cd src/nrnpython/
+        python3 setup.py install
+        cd $TRAVIS_BUILD_DIR
+        export PATH=$PATH:${install_prefix}/x86_64/bin
+        make
+        python setup.py develop
```

Comment:
does the pip installer now simplify this or we still need to build from source for MPI?

Reply:
@jasmainak I will try with pip and CircleCI. Are you using Travis still for hnn-core?

Reply:
yeah Travis is for the unit tests and CircleCI is for the documentation